use-cases/xtclim/src/anomaly.py

```diff
@@ -1,114 +1,182 @@
+            # for a better reconstruction estimate
+            train_avg_losses, _, tot_train_losses, _ = evaluate(cvae_model, trainloader,
+                                                     trainset, self.device,
+                                                    criterion,
+                                                    pixel_wise_criterion)
+            test_avg_losses, _, tot_test_losses, _ = evaluate(cvae_model, testloader,
+                                                   testset, self.device, criterion,
+                                                   pixel_wise_criterion)
             for i in range(1, n_avg):
-                proj_avg_loss, _, proj_losses, _ = evaluate(cvae_model, projloader, 
-                                                        projset, device, criterion, 
-                                                        pixel_wise_criterion)
-                tot_proj_losses = list(map(add, tot_proj_losses, proj_losses))
-                proj_avg_losses += proj_avg_loss
-            
-            tot_proj_losses = np.array(tot_proj_losses)/n_avg
-            proj_avg_losses = proj_avg_losses/n_avg
-
-            # save the losses time series
-            pd.DataFrame(tot_proj_losses).to_csv(f"../outputs/proj{scenario}_losses_{season}1d_allssp.csv")
-            print(f'SSP{scenario} Projection average loss:', proj_avg_losses, 'for', season[:-1])
\ No newline at end of file
+                train_avg_loss, _, train_losses, _ = evaluate(cvae_model, trainloader,
+                                                 trainset, self.device, criterion,
+                                                 pixel_wise_criterion)
+                tot_train_losses = list(map(add, tot_train_losses, train_losses))
+                train_avg_losses += train_avg_loss
+                test_avg_loss, _, test_losses, _ = evaluate(cvae_model, testloader,
+                                                 testset, self.device, criterion,
+                                                 pixel_wise_criterion)
+                tot_test_losses = list(map(add, tot_test_losses, test_losses))
+                test_avg_losses += test_avg_loss
+            tot_train_losses = np.array(tot_train_losses)/n_avg
+            tot_test_losses = np.array(tot_test_losses)/n_avg
+            train_avg_losses = train_avg_losses/n_avg
+            test_avg_losses = test_avg_losses/n_avg
+
+            pd.DataFrame(tot_train_losses).to_csv(f"outputs/train_losses_{season}1d_allssp.csv")
+            pd.DataFrame(tot_test_losses).to_csv(f"outputs/test_losses_{season}1d_allssp.csv")
+            print('Train average loss:', train_avg_losses)
+            print('Test average loss:', test_avg_losses)
+
+            self.log(train_avg_losses,
+                    'Average train loss',
+                    kind='metric')
+            self.log(test_avg_losses,
+                     'Average test loss',
+                     kind='metric')
```

Comment:
These look like they would fit in one line, but if they don't then you should consider the earlier comments about newlines and opening and closing brackets.